### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by jkdelauney**
Code tests added.

## Summary by Sourcery

Introduce a full testing infrastructure with pytest, refactor core filtering code for better type safety and ID handling, pin dependencies for reproducible builds, and enrich project documentation with development, testing, and AI assistant guidance

New Features:
- Add a comprehensive pytest-based test suite covering unit tests, integration tests, CLI tests, splits processing, and edge-case scenarios

Enhancements:
- Refactor filterer.py to use explicit typing for remote feed, unify variable names, and normalize entry ID handling
- Update requirements.txt to pin core dependencies and introduce requirements-dev.txt for development and testing
- Update README.md with Python requirements, development and testing instructions, and AI assistant guidance

Build:
- Add pytest.ini to configure test discovery, coverage reporting, and enforce minimum coverage thresholds

Documentation:
- Add AGENT.md for AI assistant documentation and EDGE_CASE_TESTS.md summarizing implemented edge-case tests

Tests:
- Add tests directory with conftest.py fixtures, static test data, and multiple test modules covering config loading, filtering logic, process_feed integration, and feedparser mocking

### CODE REVIEW

File: podfeedfilter/filterer.py [hunk 1]
```diff
@@ -51,11 +52,14 @@ def process_feed(cfg: FeedConfig):
...
-            existing_entries.append(e)
-            existing_ids.add(e.get('id') or e.get('link'))
+        for entry in parsed.entries:
+            existing_entries.append(entry)
+            entry_id = str(entry.get('id') or entry.get('link'))
+            existing_ids.add(entry_id)
+#           existing_ids.add(e.get('id') or e.get('link'))
 
     remote = feedparser.parse(cfg.url)
+    remote_feed = cast(feedparser.FeedParserDict, remote.feed)
```
Comment (jkdelauney) [inline podfeedfilter/filterer.py:62]: Explicit cast to resolve linting issue.

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+            'http://test/feed5': 'future_episodes_feed.xml',
+            'http://test/complex': 'complex_feed.xml',
+        }
+        
+        # Check if this is a test URL
+        if isinstance(url_or_file, str) and url_or_file in test_url_mapping:
+            xml_filename = test_url_mapping[url_or_file]
+            xml_file_path = FEED_FILES_DIR / xml_filename
+            
+            if xml_file_path.exists():
```
[the remark this answers is not in the document]
Comment (jkdelauney) [reply tests/conftest.py:297]: Will review in future refactor
